app/src/main/java/com/yterletskyi/happyfriend/features/pin/ui/PinViewModel.kt

```diff
@@ -28,6 +36,26 @@ class PinViewModel : ViewModel() {
         } catch (e: PinSizeExceededException) {
             // just ignore
         }
+        if (pin.size == PIN_CODE_MAX_LENGTH) {
+            authorize()
+        }
+    }
+
+    private fun authorize() {
+        if (pinCodeController.getPinCode() == null) {
+            pinCodeController.savePinCode(pin)
+            directionsData.value = PinFragmentDirections.toPinScreen(R.string.pin_repeat_title)
+        } else {
+            when (pinCodeController.getPinCode()?.getPinCode()) {
+                pin.getPinCode() -> {
+                    directionsData.value = PinFragmentDirections.toFriendScreen()
+                }
+                else -> {
+                    errorLiveData.value = R.string.pin_enter_error
+                    pin.clear()
```

Comment:
I agree that we have clear the `pin` here. But we also need to clear the progress in fragment - so user understand that they have to try again, right?